spec/activerecord-bitemporal/bitemporal_spec.rb

```diff
@@ -2289,6 +2289,143 @@ class EmployeeWithUniquness < Employee
+              expect(Employee.ignore_bitemporal_datetime.to_sql).not_to match %r/"transaction_to" > /
+              Employee.ignore_bitemporal_datetime.first
+            }
+            expect(result.valid_datetime).to be_nil
+            expect(result.transaction_datetime).to be_nil
+          }
+        end
+      end
+    end
+
+    describe "ActiveRecord::Bitemporal.bitemporal_at!" do
+      context "instance object" do
+        let!(:company) { Company.create(name: "Company") }
+        before do
+          emp = company.employees.create(name: "Jane")
+          emp.create_address(name: "Address")
+        end
+        it do
+          Company.includes(:employees, employees: :address).bitemporal_at(time_current).find_by(bitemporal_id: company.id).tap { |c|
+            expect(c.valid_datetime).to eq time_current
+            expect(c.employees.first.valid_datetime).to eq time_current
+            expect(c.employees.first.address.valid_datetime).to eq time_current
+            expect(c.transaction_datetime).to eq time_current
+            expect(c.employees.first.transaction_datetime).to eq time_current
+            expect(c.employees.first.address.transaction_datetime).to eq time_current
+            ActiveRecord::Bitemporal.bitemporal_at!("2019/1/1") {
+              expect(c.valid_datetime).to eq "2019/1/1"
+              expect(c.employees.first.valid_datetime).to eq "2019/1/1"
+              expect(c.employees.first.address.valid_datetime).to eq "2019/1/1"
+              expect(c.transaction_datetime).to eq "2019/1/1"
+              expect(c.employees.first.transaction_datetime).to eq "2019/1/1"
+              expect(c.employees.first.address.transaction_datetime).to eq "2019/1/1"
+            }
+          }
+        end
+
+        context "with ignore_bitemporal_datetime" do
+          it do
+            result = ActiveRecord::Bitemporal.bitemporal_at!("2019/1/1") {
+              expect(Employee.ignore_bitemporal_datetime.to_sql).not_to match %r/"valid_from" <= /
+              expect(Employee.ignore_bitemporal_datetime.to_sql).not_to match %r/"valid_to" > /
+              expect(Employee.ignore_bitemporal_datetime.to_sql).not_to match %r/"transaction_from" <= /
+              expect(Employee.ignore_bitemporal_datetime.to_sql).not_to match %r/"transaction_to" > /
+              expect(Employee.ignore_bitemporal_datetime.first.valid_datetime).to eq "2019/1/1"
+              expect(Employee.ignore_bitemporal_datetime.first.transaction_datetime).to eq "2019/1/1"
+              Employee.ignore_bitemporal_datetime.first
+            }
+            expect(result.valid_datetime).to be_nil
+            expect(result.transaction_datetime).to be_nil
+          end
+        end
+      end
+
+      context "relation object" do
+        # TODO: Cannot overwrite valid_datetime/transaction_datetime with bitemporal_at
+        xit do
+            Company.bitemporal_at("2019/1/1").tap { |m|
```

Comment:
[nits] The indentation seems to be misaligned.
```suggestion
          Company.bitemporal_at("2019/1/1").tap { |m|
```